sysfont loading silently fails when `fc-list` is missing, `get_fonts` returns `[None]` instead of `[]`

**Environment:**

You can get some of this info from the text that pops up in the console when you run a pygame program.

- **Operating system in the WSL**: Ubuntu 20.04 (Linux version 5.10.16.3-microsoft-standard-WSL2)
- **Operating system in the host**: Windows 10 Pro 21H1 (OS Build 19043.1526)
- **Python version**: 3.8.10
- **SDL version**: 2.0.10
- **FreeType version**: 23.1.17
- **PyGame version**: 2.1.3.dev5 (Last commit in the develop)

**Current behavior:**

pygame.tests.font_test, pygame.tests.ftfont_test and pygame.tests.sysfont_test run with an error under WSL on Windows 10. 
(see *Test errors* below)

**Expected behavior:**

```bash
$ sudo SDL_VIDEODRIVER=dummy SDL_AUDIODRIVER=disk venv2/bin/python3 -m pygame.tests --exclude opengl
...
...
...
----------------------------------------------------------------------
Ran 1673 tests in ~30 seconds

OK
```

**Steps to reproduce:**

1. Run WSL
2. git clone pygame
3. cd pygame
4. python3 setup.py build
5. Create virtual enviroment
6. python3 setup.py install
7. SDL_VIDEODRIVER=dummy SDL_AUDIODRIVER=disk python3 -m pygame.tests --exclude opengl

**Test errors**

```
======================================================================
ERROR: test_match_font_all_exist (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 119, in test_match_font_all_exist
    path = pygame_font.match_font(font)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_bold (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 171, in test_match_font_bold
    self.assertTrue(any(pygame_font.match_font(font, bold=True) for font in fonts))
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 171, in <genexpr>
    self.assertTrue(any(pygame_font.match_font(font, bold=True) for font in fonts))
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_italic (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 178, in test_match_font_italic
    any(pygame_font.match_font(font, italic=True) for font in fonts)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 178, in <genexpr>
    any(pygame_font.match_font(font, italic=True) for font in fonts)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_name (pygame.tests.font_test.FontModuleTest)
That match_font accepts names of various types
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 126, in test_match_font_name
    font_path = pygame_font.match_font(font)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
FAIL: test_get_fonts (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 99, in test_get_fonts
    self.assertTrue(isinstance(name, str), name)
AssertionError: False is not true

======================================================================
ERROR: test_match_font_all_exist (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 119, in test_match_font_all_exist
    path = pygame_font.match_font(font)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_bold (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 171, in test_match_font_bold
    self.assertTrue(any(pygame_font.match_font(font, bold=True) for font in fonts))
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 171, in <genexpr>
    self.assertTrue(any(pygame_font.match_font(font, bold=True) for font in fonts))
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_italic (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 178, in test_match_font_italic
    any(pygame_font.match_font(font, italic=True) for font in fonts)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 178, in <genexpr>
    any(pygame_font.match_font(font, italic=True) for font in fonts)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_name (pygame.tests.ftfont_test.FtFontModuleTest)
That match_font accepts names of various types
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 126, in test_match_font_name
    font_path = pygame_font.match_font(font)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
FAIL: test_get_fonts (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 99, in test_get_fonts
    self.assertTrue(isinstance(name, str), name)
AssertionError: False is not true

======================================================================
FAIL: test_create_aliases (pygame.tests.sysfont_test.SysfontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/sysfont_test.py", line 11, in test_create_aliases
    self.assertTrue(len(pygame.sysfont.Sysalias) > 0)
AssertionError: False is not true
```

**Terminal log**

[Full log (1410 lines)](https://pastebin.com/riqZzsTm)

Abridged log:
```bash
quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ git pull; date
Already up to date.
Thu Apr 28 14:50:27 MSK 2022
quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ sudo python3 setup.py build


WARNING, No "Setup" File Exists, Running "buildconfig/config.py"
Using UNIX configuration...


Hunting dependencies...
SDL     : found 2.0.10
FONT    : found
IMAGE   : found
MIXER   : found
PNG     : found
JPEG    : found
SCRAP   : found
PORTMIDI: found
PORTTIME: found
FREETYPE: found 23.1.17

If you get compiler errors during install, double-check
the compiler flags in the "Setup" file.


Continuing With "setup.py"
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.8
creating build/lib.linux-x86_64-3.8/pygame
copying src_py/camera.py -> build/lib.linux-x86_64-3.8/pygame
copying src_py/colordict.py -> build/lib.linux-x86_64-3.8/pygame
Removed so as not to increase the logs
running build_ext
building 'pygame.imageext' extension
creating build/temp.linux-x86_64-3.8
creating build/temp.linux-x86_64-3.8/src_c
x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -D_REENTRANT -I/usr/X11R6/include -I/usr/include/SDL2 -I/usr/include/SDL2 -I/usr/include -I/usr/include -I/usr/include/python3.8 -c src_c/imageext.c -o build/temp.linux-x86_64-3.8/src_c/imageext.o -Wall -Wno-error=unknown-pragmas
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.8/src_c/imageext.o -lSDL2 -lSDL2_image -lpng -ljpeg -o build/lib.linux-x86_64-3.8/pygame/imageext.cpython-38-x86_64-linux-gnu.so
building 'pygame.font' extension
x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -D_REENTRANT -I/usr/X11R6/include -I/usr/include/SDL2 -I/usr/include/SDL2 -I/usr/include/python3.8 -c src_c/font.c -o build/temp.linux-x86_64-3.8/src_c/font.o -Wall -Wno-error=unknown-pragmas
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.8/src_c/font.o -lSDL2 -lSDL2_ttf -o build/lib.linux-x86_64-3.8/pygame/font.cpython-38-x86_64-linux-gnu.so
building 'pygame.mixer' extension
x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -D_REENTRANT -I/usr/X11R6/include -I/usr/include/SDL2 -I/usr/include/SDL2 -I/usr/include/python3.8 -c src_c/mixer.c -o build/temp.linux-x86_64-3.8/src_c/mixer.o -Wall -Wno-error=unknown-pragmas
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.8/src_c/mixer.o -lSDL2 -lSDL2_mixer -o build/lib.linux-x86_64-3.8/pygame/mixer.cpython-38-x86_64-linux-gnu.so
Removed so as not to increase the logs
quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ sudo python3 -m venv venv2
quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ source venv2/bin/activate
(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ which python3
/mnt/d/PycharmProjects/pygame/venv2/bin/python3
(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ sudo venv2/bin/python3 setup.py install --prefix="./venv2/"
running install
running bdist_egg
running egg_info
writing pygame.egg-info/PKG-INFO
writing dependency_links to pygame.egg-info/dependency_links.txt
writing entry points to pygame.egg-info/entry_points.txt
writing top-level names to pygame.egg-info/top_level.txt
package init file 'docs/__init__.py' not found (or not a regular file)
reading manifest file 'pygame.egg-info/SOURCES.txt'
writing manifest file 'pygame.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
copying src_py/version.py -> build/lib.linux-x86_64-3.8/pygame
running build_ext
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/pygame
copying build/lib.linux-x86_64-3.8/pygame/base.cpython-38-x86_64-linux-gnu.so -> build/bdist.linux-x86_64/egg/pygame
copying build/lib.linux-x86_64-3.8/pygame/bufferproxy.cpython-38-x86_64-linux-gnu.so -> build/bdist.linux-x86_64/egg/pygame
copying build/lib.linux-x86_64-3.8/pygame/camera.py -> build/bdist.linux-x86_64/egg/pygame
copying build/lib.linux-x86_64-3.8/pygame/color.cpython-38-x86_64-linux-gnu.so -> build/bdist.linux-x86_64/egg/pygame
Removed so as not to increase the logs
byte-compiling build/bdist.linux-x86_64/egg/pygame/camera.py to camera.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/pygame/colordict.py to colordict.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/pygame/cursors.py to cursors.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/pygame/docs/__main__.py to __main__.cpython-38.pyc
Removed so as not to increase the logs
creating stub loader for pygame/imageext.cpython-38-x86_64-linux-gnu.so
creating stub loader for pygame/font.cpython-38-x86_64-linux-gnu.so
creating stub loader for pygame/mixer.cpython-38-x86_64-linux-gnu.so
creating stub loader for pygame/mixer_music.cpython-38-x86_64-linux-gnu.so
Removed so as not to increase the logs
byte-compiling build/bdist.linux-x86_64/egg/pygame/imageext.py to imageext.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/pygame/font.py to font.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/pygame/mixer.py to mixer.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/pygame/mixer_music.py to mixer_music.cpython-38.pyc
Removed so as not to increase the logs
installing package data to build/bdist.linux-x86_64/egg
running install_data
copying buildconfig/stubs/pygame/py.typed -> build/bdist.linux-x86_64/egg/pygame
copying buildconfig/stubs/pygame/base.pyi -> build/bdist.linux-x86_64/egg/pygame
copying buildconfig/stubs/pygame/bufferproxy.pyi -> build/bdist.linux-x86_64/egg/pygame
copying buildconfig/stubs/pygame/camera.pyi -> build/bdist.linux-x86_64/egg/pygame
Removed so as not to increase the logs
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying pygame.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying pygame.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying pygame.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying pygame.egg-info/entry_points.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying pygame.egg-info/not-zip-safe -> build/bdist.linux-x86_64/egg/EGG-INFO
copying pygame.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
writing build/bdist.linux-x86_64/egg/EGG-INFO/native_libs.txt
creating 'dist/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing pygame-2.1.3.dev5-py3.8-linux-x86_64.egg
removing '/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg' (and everything under it)
creating /mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg
Extracting pygame-2.1.3.dev5-py3.8-linux-x86_64.egg to /mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages
Adding pygame 2.1.3.dev5 to easy-install.pth file

Installed /mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg
Processing dependencies for pygame==2.1.3.dev5
Finished processing dependencies for pygame==2.1.3.dev5
(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ python3 -c "import pygame; print(pygame.__version__)"
pygame 2.1.3.dev5 (SDL 2.0.10, Python 3.8.10)
Hello from the pygame community. https://www.pygame.org/contribute.html
2.1.3.dev5
(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ sudo SDL_VIDEODRIVER=dummy SDL_AUDIODRIVER=disk venv2/bin/python3 -m pygame.tests --exclude opengl
pygame 2.1.3.dev5 (SDL 2.0.10, Python 3.8.10)
Hello from the pygame community. https://www.pygame.org/contribute.html
skipping pygame.tests.scrap_test (tag 'ignore')
skipping pygame.tests.sndarray_test (tag 'ignore')
skipping pygame.tests.surfarray_test (tag 'ignore')
loading pygame.tests.base_test
loading pygame.tests.blit_test
loading pygame.tests.bufferproxy_test
loading pygame.tests.camera_test
loading pygame.tests.color_test
loading pygame.tests.constants_test
loading pygame.tests.context_test
loading pygame.tests.controller_test
loading pygame.tests.cursors_test
loading pygame.tests.display_test
loading pygame.tests.docs_test
loading pygame.tests.draw_test
loading pygame.tests.event_test
loading pygame.tests.font_test
loading pygame.tests.freetype_test
loading pygame.tests.ftfont_test
loading pygame.tests.gfxdraw_test
loading pygame.tests.image__save_gl_surface_test
loading pygame.tests.image_test
loading pygame.tests.imageext_test
loading pygame.tests.joystick_test
loading pygame.tests.key_test
loading pygame.tests.locals_test
loading pygame.tests.mask_test
loading pygame.tests.math_test
loading pygame.tests.midi_test
loading pygame.tests.mixer_music_test
loading pygame.tests.mouse_test
loading pygame.tests.pixelarray_test
loading pygame.tests.pixelcopy_test
loading pygame.tests.rect_test
loading pygame.tests.rwobject_test
loading pygame.tests.sprite_test
loading pygame.tests.surface_test
CRITICAL: You are using the SDL disk i/o audio driver!
CRITICAL:  Writing to file [sdlaudio.raw].
loading pygame.tests.surflock_test
loading pygame.tests.sysfont_test
loading pygame.tests.threads_test
loading pygame.tests.time_test
CRITICAL: You are using the SDL disk i/o audio driver!
CRITICAL:  Writing to file [sdlaudio.raw].
loading pygame.tests.touch_test
loading pygame.tests.transform_test
loading pygame.tests.version_test
loading pygame.tests.video_test
...........................................................................................................................................sssssss.......s....s..ss..........ss.....ss..........ss..............................................................................................................................................................................................................................................s................s.......F......EEEE............................F......EEEE......................................................................................s.....s........s.....................................................................................................................................................xxxxxxx......x...........x........x.............................................x...x..............................................................................................................................................................................................s................s.................................s.....s.s...............................................................................................................................................................................................................................................................................................................................................................................................................s.................................................................s..s...s................................F.s.s......................s.s.............................................
======================================================================
ERROR: test_match_font_all_exist (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 119, in test_match_font_all_exist
    path = pygame_font.match_font(font)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_bold (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 171, in test_match_font_bold
    self.assertTrue(any(pygame_font.match_font(font, bold=True) for font in fonts))
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 171, in <genexpr>
    self.assertTrue(any(pygame_font.match_font(font, bold=True) for font in fonts))
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_italic (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 178, in test_match_font_italic
    any(pygame_font.match_font(font, italic=True) for font in fonts)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 178, in <genexpr>
    any(pygame_font.match_font(font, italic=True) for font in fonts)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_name (pygame.tests.font_test.FontModuleTest)
That match_font accepts names of various types
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 126, in test_match_font_name
    font_path = pygame_font.match_font(font)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
FAIL: test_get_fonts (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 99, in test_get_fonts
    self.assertTrue(isinstance(name, str), name)
AssertionError: False is not true

======================================================================
ERROR: test_match_font_all_exist (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 119, in test_match_font_all_exist
    path = pygame_font.match_font(font)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_bold (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 171, in test_match_font_bold
    self.assertTrue(any(pygame_font.match_font(font, bold=True) for font in fonts))
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 171, in <genexpr>
    self.assertTrue(any(pygame_font.match_font(font, bold=True) for font in fonts))
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_italic (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 178, in test_match_font_italic
    any(pygame_font.match_font(font, italic=True) for font in fonts)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 178, in <genexpr>
    any(pygame_font.match_font(font, italic=True) for font in fonts)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_match_font_name (pygame.tests.ftfont_test.FtFontModuleTest)
That match_font accepts names of various types
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 126, in test_match_font_name
    font_path = pygame_font.match_font(font)
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/sysfont.py", line 483, in match_font
    for single_name in name:
TypeError: 'NoneType' object is not iterable

======================================================================
FAIL: test_get_fonts (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/font_test.py", line 99, in test_get_fonts
    self.assertTrue(isinstance(name, str), name)
AssertionError: False is not true

======================================================================
FAIL: test_create_aliases (pygame.tests.sysfont_test.SysfontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pygame-2.1.3.dev5-py3.8-linux-x86_64.egg/pygame/tests/sysfont_test.py", line 11, in test_create_aliases
    self.assertTrue(len(pygame.sysfont.Sysalias) > 0)
AssertionError: False is not true

----------------------------------------------------------------------
Ran 1673 tests in 29.668s

FAILED (failures=3, errors=8)

(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ python3 --version
Python 3.8.10
(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ pip3 --version
pip 20.0.2 from /mnt/d/PycharmProjects/pygame/venv2/lib/python3.8/site-packages/pip (python 3.8)
(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04 LTS
Release:        20.04
Codename:       focal
(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ cat /proc/version
Linux version 5.10.16.3-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Fri Apr 2 22:23:49 UTC 2021
(venv2) quswadress-ubuntu@DESKTOP-HUJN9K4:/mnt/d/PycharmProjects/pygame$ gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

*P.S Forgive my English. I'm still learning it.*

--- a/src_py/sysfont.py
+++ b/src_py/sysfont.py
@@ -20,28 +20,19 @@
 """sysfont, used in the font module to find system fonts"""
 
 import os
+import subprocess
 import sys
+import warnings
 from os.path import basename, dirname, exists, join, splitext
 
 from pygame.font import Font
-
 
 OpenType_extensions = frozenset((".ttf", ".ttc", ".otf"))
 Sysfonts = {}
 Sysalias = {}
 
-# Python 3 compatibility
-
-
-def toascii(raw):
-    """convert bytes to ASCII-only string"""
-    return raw.decode("ascii", "ignore")
-
-
 if os.name == "nt":
     import winreg as _winreg
-else:
-    import subprocess
 
 
 def _simplename(name):
@@ -204,34 +195,38 @@ def initsysfonts_unix(path="fc-list"):
     fonts = {}
 
     try:
-        # pylint: disable=consider-using-with
-        # subprocess.Popen is not a context manager in all of
-        # pygame's supported python versions.
+        proc = subprocess.run(
+            [path, ":", "file", "family", "style"],
+            stdout=subprocess.PIPE,  # capture stdout
+            stderr=subprocess.PIPE,  # capture stderr
+            check=True,  # so that errors raise python exception which is handled below
+            timeout=1,  # so that we don't hang the program waiting
+        )
 
-        # note, we capture stderr so if fc-list isn't there to stop stderr
-        # printing.
-        flout, _ = subprocess.Popen(
-            f"{path} : file family style",
-            shell=True,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            close_fds=True,
-        ).communicate()
-    except (OSError, ValueError):
-        return fonts
+    except FileNotFoundError:
+        warnings.warn(
+            f"'{path}' is missing, system fonts cannot be loaded on your platform"
+        )
 
-    entries = toascii(flout)
-    try:
-        for entry in entries.split("\n"):
+    except subprocess.TimeoutExpired:
+        warnings.warn(
+            f"Process running '{path}' timed-out! System fonts cannot be loaded on "
+            "your platform"
+        )
 
+    except subprocess.CalledProcessError as e:
+        warnings.warn(
+            f"'{path}' failed with error code {e.returncode}! System fonts cannot be "
+            f"loaded on your platform. Error log is:\n{e.stderr}"
+        )
+
+    else:
+        for entry in proc.stdout.decode("ascii", "ignore").splitlines():
             try:
                 _parse_font_entry_unix(entry, fonts)
             except ValueError:
                 # try the next one.
                 pass
-
-    except ValueError:
-        pass
 
     return fonts
 
@@ -458,7 +453,9 @@ def get_fonts():
     """
     if not Sysfonts:
         initsysfonts()
-    return list(Sysfonts)
+
+    # 'Sysfonts' can contain a sentinel 'None' key, don't forward that to users
+    return [i for i in Sysfonts if i is not None]
 
 
 def match_font(name, bold=0, italic=0):

--- a/src_py/sysfont.py
+++ b/src_py/sysfont.py
@@ -31,6 +31,8 @@ OpenType_extensions = frozenset((".ttf", ".ttc", ".otf"))
 Sysfonts = {}
 Sysalias = {}
 
+is_init = False
+
 if os.name == "nt":
     import winreg as _winreg
 
@@ -332,6 +334,11 @@ def initsysfonts():
 
     Has different initialisation functions for different platforms.
     """
+    global is_init
+    if is_init:
+        # no need to re-init
+        return
+
     if sys.platform == "win32":
         fonts = initsysfonts_win32()
     elif sys.platform == "darwin":
@@ -340,8 +347,7 @@ def initsysfonts():
         fonts = initsysfonts_unix()
     Sysfonts.update(fonts)
     create_aliases()
-    if not Sysfonts:  # dummy so we don't try to reinit
-        Sysfonts[None] = None
+    is_init = True
 
 
 def font_constructor(fontpath, size, bold, italic):
@@ -394,8 +400,7 @@ def SysFont(name, size, bold=False, italic=False, constructor=None):
     if constructor is None:
         constructor = font_constructor
 
-    if not Sysfonts:
-        initsysfonts()
+    initsysfonts()
 
     gotbold = gotitalic = False
     fontname = None
@@ -451,11 +456,8 @@ def get_fonts():
     removed. This is how pygame internally stores the font
     names for matching.
     """
-    if not Sysfonts:
-        initsysfonts()
-
-    # 'Sysfonts' can contain a sentinel 'None' key, don't forward that to users
-    return [i for i in Sysfonts if i is not None]
+    initsysfonts()
+    return list(Sysfonts)
 
 
 def match_font(name, bold=0, italic=0):
@@ -470,8 +472,7 @@ def match_font(name, bold=0, italic=0):
 
     If no match is found, None is returned.
     """
-    if not Sysfonts:
-        initsysfonts()
+    initsysfonts()
 
     fontname = None
     if isinstance(name, (str, bytes)):

